documentation/src/main/asciidoc/user_guide/endpoint_interop.adoc

```diff
@@ -0,0 +1,154 @@
+==  Embedded/Remote Interoperability[[endpoint.interop]]
+
+
+{brandname} offers the possibility to store and retrieve data in a local embedded way, and also remotely thanks to multiple endpoints offered.
+
+When configuring the cache with the <<encoding:media_type, MediaType>> of the data to be stored, users can read and write data in multiple formats taking advantage of the internal support for data conversion.
+
+Depending on the level of interoperability required of a single cache being accessed from multiple endpoints by different clients, a specific storage format must be configured, clients may need to use different strategies when interacting with cache or even custom classes may be required to be deployed in the server.
+
+
+=== Planning for interoperability
+
+In general REST clients are better suitable to deal with text formats such as JSON, XML or plain text instead of binary, although the REST endpoint does support <<rest.key_content_type, representing binary values in hexadecimal or base64 format>>.
+
+Java Hot Rod clients usually deal with objects representing the entities stored in the cache, and through the marshalling operation, will have those entities mapped to/from bytes to be sent across the wire.
+
+Similarly, non-java Hot Rod clients (C++/C#/Javascript) are also better suited to deal with native objects in those languages. The caveat is, if they must interoperate with Java clients, a platform independent format must be chosen.
+
+
+=== Basic Interoperability between REST/Hot Rod using text
+
+If querying/indexing is not required, and the cache will store only text, it's enough to configure the cache keys and values MediaType with `text/plain; charset=UTF-8` (or any other charset) or another text based format such as JSON and XML (optionally carrying a custom charset).
+
+The Hot Rod client must be configured with a `org.infinispan.commons.marshall.StringMarshaller`, and the REST clients must send the correct headers when writing:
+
+ Content-Type: text/plain; charset=UTF-8
+
+and reading:
+
+ Accept: text/plain; charset=UTF-8 (or any other charset)
+
+In summary:
+
+|===
+|REST clients|Java Hot Rod clients|Query|Non Java Hot Rod clients|Custom objects|Avoid Class deployment
+|[green]*Y*|[green]*Y* |[red]*N*|[red]*N*|[red]*N* |[green]*Y*
+|===
+
+
+=== Interoperability between REST/Hot Rod using custom Java objects
+
+If storage is done using a marshalled, custom java object (this is the default behaviour when using the Java Hot Rod client) and no query is required, it's necessary to configure the cache with the MediaType of the marshalled storage. By default, the Java Hot Rod client uses JBoss marshalling format to send data to the server and the entries get stored in this format.
+
+[source,xml]
+----
+<distributed-cache name="my-cache">
+   <encoding>
+      <key media-type="application/x-jboss-marshalling"/>
```

Comment:
Is this configuration required? If so, in which specific situation?

Reply:
Generally speaking, configuring the media type is required if you want to read data in multiple formats, as otherwise the server will need to guess the format. This specific case is required if you'll have REST clients reading data as JSON

Reply:
Ok, it's unclear in the documentation. In other cases you've explained when to use certain configuration, but here it was missing.

Reply:
I will clarify it